### PR TITLE
integ-tests: test security groups are properly applied to a cluster

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
1. Test `additional_sg` in the config file is added to head and compute nodes
2. Test `ssh_from` in the config file applies to the pcluster security group of the head node
3. Test `vpc_security_group_id` in the config file overwrites security group of head and compute nodes, FSx, and EFS

Signed-off-by: Hanwen <hanwenli@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/networking/test_security_groups.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        mount_target_security_groups = efs_client.describe_mount_target_security_groups(MountTargetId=mount_target_id)[
+            "SecurityGroups"
+        ]
+        assert_that(mount_target_security_groups[0]).is_equal_to(custom_security_group_id)
+        assert_that(len(mount_target_security_groups) == 1).is_true()
+
+
+@pytest.fixture(scope="class")
+def custom_security_group(vpc_stack, region, request, cfn_stacks_factory):
+    # efs stack contains a single efs resource
```
Comment (demartinofra) [inline tests/integration-tests/tests/networking/test_security_groups.py:100]: probably it's a leftover from a copy/paste

File: tests/integration-tests/tests/networking/test_security_groups/test_additional_sg_and_ssh_from/pcluster.config.ini [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+aws_region_name = {{ region }}
+
+[cluster default]
+base_os = {{ os }}
+key_name = {{ key_name }}
+vpc_settings = parallelcluster-vpc
+scheduler = {{ scheduler }}
+master_instance_type = {{ instance }}
+compute_instance_type = {{ instance }}
+initial_queue_size = 1
```
Comment (demartinofra) [inline tests/integration-tests/tests/networking/test_security_groups/test_additional_sg_and_ssh_from/pcluster.config.ini:14]: initial and max queue_size are not applied with awsbatch. We need to use min_vcpus. Same in the other config

File: tests/integration-tests/tests/networking/test_security_groups.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    The additional security group should be added to the head and compute nodes. The
+    """
+    custom_security_group_id = custom_security_group.cfn_resources["SecurityGroupResource"]
+    ssh_from = "10.11.12.0/32"
+    cluster_config = pcluster_config_reader(additional_sg=custom_security_group_id, ssh_from=ssh_from)
+    cluster = clusters_factory(cluster_config)
+    ec2_client = boto3.client("ec2", region_name=region)
+    instances = _get_instances_by_security_group(ec2_client, custom_security_group_id)
+    logging.info("Asserting that head node and compute node has the additional security group")
+    assert_that(len(instances)).is_equal_to(2)
```
Comment (demartinofra) [inline tests/integration-tests/tests/networking/test_security_groups.py:37]: nit: you can use assert_that().is_length(2). Same everywhere else you assert the length

**Review by demartinofra — COMMENTED** [review]
Well done! LGTM. Only dropped a few minor comments

**Review by demartinofra — DISMISSED** [review]

File: tests/integration-tests/configs/develop.yaml [hunk 4]
```diff
@@ -4,6 +4,27 @@ test-suites:
 {% filter indent(2) %}
 {% include 'common/common.yaml' %}
 {% endfilter %}
+  networking:
```
Comment (demartinofra) [inline tests/integration-tests/configs/develop.yaml:7]: let's add the new tests in common.yaml unless we want to exclude them from release tests

Comment (hanwen-pcluste) [reply tests/integration-tests/configs/develop.yaml]: Done

**Review by demartinofra — APPROVED** [review]

**Pull request merged**